contracts/src/TroveManager.sol

```diff
@@ -464,6 +464,18 @@ contract TroveManager is LiquityBase, ITroveManager, ITroveEvents {
         return _status == Status.active || _status == Status.zombie;
     }
 
+    // Troves with debt adjusted in the same block cannot be liquidated
+    // This is to prevent attacks involving sandwiching a price update to self-liquidate
+    function _isRecent(uint256 _troveId) internal view returns (bool) {
+        if (uint256(Troves[_troveId].lastDebtUpdateTime) >= block.timestamp) return true;
```

Comment:
An individual Trove's `lastDebtUpdateTime` is updated when they open or adjust debt/coll but also when they adjust their interest rate.

IIRC, interest rate adjustments don't actually enforce that the resulting ICR >= MCR (or do they - am I missing that?) So what if a Trove at ICR < MCR keeps adjusting interest rate every block - can they avoid liquidation indefinitely?

Reply:
I would say we should enforce MCR on adjusting interest. Not only for this, but also because it helps system stability.
The more we force troves under MCR to go back above it the better in my opinion.

Reply:
It is actually enforced, in `_applyUpfrontFee` function.

Reply:
See:
https://github.com/liquity/bold/blob/6e61065a3cbe7939f7f170614376774bba50df5e/contracts/src/BorrowerOperations.sol#L535
https://github.com/liquity/bold/blob/6e61065a3cbe7939f7f170614376774bba50df5e/contracts/src/BorrowerOperations.sol#L1134

Reply:
Ah cool, thanks.  That would stop indefinite individual griefing at least - adjusting every block would incur premature fees and thus this check then.